### DISCUSSION TITLE
nms operators supporting onnx

### DISCUSSION THREAD

**Discussion by yxdragon** [ultralytics/ultralytics#25246] [Show and tell]
### Search before asking

- [x] I have searched the Ultralytics [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar feature requests.

### Description

Hi, I wrote nms in torch, that supporting onnx. https://github.com/VectorElectron/yolov11/blob/main/yolov11_vectron/operators/nms.py

[github](https://github.com/VectorElectron/yolov11)
[live demo of onnxruntime-web](https://vectorelectron.github.io/release/yolov11/index.html)

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

**Comment by UltralyticsAssistant** [comment]
👋 Hello @yxdragon, thank you for sharing your ONNX-compatible NMS implementation and demo 🚀! This is an automated response, and an Ultralytics engineer will also assist you soon.

We recommend visiting the [Docs](https://docs.ultralytics.com/) for new users, where you can find many [Python](https://docs.ultralytics.com/usage/python) and [CLI](https://docs.ultralytics.com/usage/cli) usage examples and answers to common questions.

If this is a 🐛 bug report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example) to help us investigate it. For an ONNX-related request, please also include the Ultralytics package version, model and export arguments, target ONNX opset, runtime and hardware, complete logs, and a small example demonstrating the expected and actual behavior.

If this is a custom training ❓ question, please provide as much information as possible, including dataset image examples and training logs, and verify that you are following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips).

Join the Ultralytics community where it suits you best. For real-time chat, head to [Discord](https://discord.com/invite/ultralytics) 🎧. Prefer in-depth discussions? Check out [Discourse](https://community.ultralytics.com/). Or dive into threads on our [Subreddit](https://www.reddit.com/r/Ultralytics/) to share knowledge with the community.

## Upgrade

Upgrade to the latest `ultralytics` package, including all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml), in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/) to verify whether the behavior is already resolved:

```bash
pip install -U ultralytics
```

## Environments

YOLO may be run in any of the following up-to-date verified environments, with all dependencies including [CUDA](https://developer.nvidia.com/cuda)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled:

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo26"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes) and [Tasks](https://docs.ultralytics.com/tasks) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

**Comment by glenn-jocher** [comment]
Thanks for sharing. Ultralytics already supports embedded NMS during ONNX export with `model.export(format="onnx", nms=True)` via [`NMSModel`](https://docs.ultralytics.com/reference/engine/exporter#ultralytics.engine.exporter.NMSModel); if your implementation improves compatibility or performance for a specific runtime such as ONNX Runtime Web, please open a focused PR with tests and benchmarks demonstrating the difference.
